src/signed_data.rs

```diff
@@ -250,158 +249,72 @@ impl<'a> FromDer<'a> for SubjectPublicKeyInfo<'a> {
-
-    /// Verify a signature.
-    ///
-    /// `public_key` is the `subjectPublicKey` value from a `SubjectPublicKeyInfo` encoding
-    ///  and is untrusted.
-    ///
-    ///  `message` is the data over which the signature was allegedly computed.
-    ///  It is not hashed; implementations of this trait function must do hashing
-    ///  if that is required by the algorithm they implement.
-    ///
-    ///  `signature` is the signature allegedly over `message`.
-    ///
-    /// Return `Ok(())` only if `signature` is a valid signature on `message`.
-    ///
-    /// Return `Err(InvalidSignature)` if the signature is invalid, including if the `public_key`
-    /// encoding is invalid.  There is no need or opportunity to produce errors
-    /// that are more specific than this.
-    fn verify_signature(
-        &self,
-        public_key: &[u8],
-        message: &[u8],
-        signature: &[u8],
-    ) -> Result<(), InvalidSignature>;
-}
-
-/// A detail-less error when a signature is not valid.
-#[derive(Debug, Copy, Clone)]
-pub struct InvalidSignature;
-
-/// Encodings of the PKIX AlgorithmIdentifier type:
-///
-/// ```ASN.1
-/// AlgorithmIdentifier  ::=  SEQUENCE  {
-///     algorithm               OBJECT IDENTIFIER,
-///     parameters              ANY DEFINED BY algorithm OPTIONAL  }
-///                                -- contains a value of the type
-///                                -- registered for use with the
-///                                -- algorithm object identifier value
-/// ```
-/// (from <https://www.rfc-editor.org/rfc/rfc5280#section-4.1.1.2>)
-///
-/// The outer sequence encoding is not included, so this is an OID encoding
-/// for `algorithm` plus the `parameters` value.
+/// Encodings of the PKIX AlgorithmIdentifier type.
 ///
 /// This module contains a set of common values, and exists to keep the
 /// names of these separate from the actual algorithm implementations.
 pub mod alg_id {
-    /// A `AlgorithmIdentifier` encoding.
-    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-    pub struct AlgorithmIdentifier {
-        asn1_id_value: untrusted::Input<'static>,
-    }
-
-    impl AlgorithmIdentifier {
-        /// Makes a new `AlgorithmIdentifier` from a static octet string.
-        ///
-        /// This does not validate the contents of the string.
-        pub const fn new(bytes: &'static [u8]) -> Self {
-            Self {
-                asn1_id_value: untrusted::Input::from(bytes),
-            }
-        }
-
-        pub(crate) fn matches_algorithm_id_value(&self, encoded: untrusted::Input) -> bool {
-            encoded == self.asn1_id_value
-        }
-    }
+    use super::AlgorithmIdentifier;
 
     // See src/data/README.md.
 
     /// AlgorithmIdentifier for `id-ecPublicKey` with named curve `secp256r1`.
-    pub const ECDSA_P256: AlgorithmIdentifier =
-        AlgorithmIdentifier::new(include_bytes!("data/alg-ecdsa-p256.der"));
+    pub static ECDSA_P256: AlgorithmIdentifier =
```

Comment:
Should these be changed back to `const` now?

Reply:
oh, yes, that's an oversight